src/core/image.js

```diff
@@ -103,25 +103,31 @@ class PDFImage {
     const dict = image.dict;
 
     const filter = dict.get("F", "Filter");
+    let filterName;
     if (filter instanceof Name) {
-      switch (filter.name) {
-        case "JPXDecode":
-          const jpxImage = new JpxImage();
-          jpxImage.parseImageProperties(image.stream);
-          image.stream.reset();
-
-          image.width = jpxImage.width;
-          image.height = jpxImage.height;
-          image.bitsPerComponent = jpxImage.bitsPerComponent;
-          image.numComps = jpxImage.componentsCount;
-          break;
-        case "JBIG2Decode":
-          image.bitsPerComponent = 1;
-          image.numComps = 1;
-          break;
+      filterName = filter.name;
+    } else if (Array.isArray(filter)) {
+      const filterZero = xref.fetchIfRef(filter[0]);
```

Comment:
If I understand correctly, we're only considering the first entry in the array (and the same here https://github.com/mozilla/pdf.js/blob/14a8b819dc44feb7a0f08c8a5fd9878135f36646/src/core/parser.js#L510).
If I correctly read the specs, "Filter" can be a name or an array, and on page 216 there is this example:
![image](https://user-images.githubusercontent.com/5641725/180730643-7f4815a6-d075-4de9-99b9-b0092722d961.png)
Does it mean that other filters are just skipped ? or are they considered somewhere else in the code ? or did I wrongly read the docs ?

Reply:
> Does it mean that other filters are just skipped ? or are they considered somewhere else in the code ?

Nope, all available filters are considered (in order) when decoding the stream; please see https://github.com/mozilla/pdf.js/blob/14a8b819dc44feb7a0f08c8a5fd9878135f36646/src/core/parser.js#L692-L723

For images, which is what we're dealing with here, it should suffice to consider the *first* /Filter-entry to determine its kind.